### PR TITLE
Mostly complete new login system aside from some minor UI tweaks

### PR DESCRIPTION
**Pull request by effectsToCause**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (mderasse) [inline .vscode/settings.json]: To Remove :)

File: src/qt/askpassphrasedialog.cpp [hunk 2]
```diff
@@ -75,6 +77,20 @@ AskPassphraseDialog::AskPassphraseDialog(Mode _mode, QWidget *parent, SecureStri
...
     connect(ui->passEdit2, &QLineEdit::textChanged, this, &AskPassphraseDialog::textChanged);
     connect(ui->passEdit3, &QLineEdit::textChanged, this, &AskPassphraseDialog::textChanged);
+
+    // apply style
+    // haven't figured out how to discriminate buttons in QDialogButtonBox in style.qss yet
+     ui->buttonBox->button(QDialogButtonBox::Ok)->setStyleSheet(QString("border: no; border-radius: 3px; height: 25px; width: 75px; padding: 5px 10px; background-color: #418bca; color: white; font: 10pt Lato;font-weight: bold;"));
+     ui->buttonBox->button(QDialogButtonBox::Cancel)->setStyleSheet(QString("border: no; border-radius: 3px; height: 25px; width: 75px; padding: 5px 10px; background-color: #e93a5d; color: white; font: 10pt Lato;font-weight: bold;"));
+    // XXX: Use local path for development
+    // XXX: To remove before for release
+    QString strPath(QCoreApplication::applicationDirPath() + "/res/style.qss");
```
Comment (mderasse) [inline src/qt/askpassphrasedialog.cpp]: Do we really need to recall the style.qss ? it should be load for all the app

File: src/qt/askpassphrasedialog.cpp [hunk 3]
```diff
@@ -113,7 +129,7 @@ void AskPassphraseDialog::accept()
             break;
         }
         QMessageBox::StandardButton retval = QMessageBox::question(this, tr("Confirm wallet encryption"),
-                 tr("Warning: If you encrypt your wallet and lose your passphrase, you will <b>LOSE ALL OF YOUR BITCOINS</b>!") + "<br><br>" + tr("Are you sure you wish to encrypt your wallet?"),
```
Comment (mderasse) [inline src/qt/askpassphrasedialog.cpp:116]: Nice catch ;)

File: src/qt/forms/askpassphrasedialog.ui [hunk 4]
```diff
@@ -7,7 +7,7 @@
     <x>0</x>
     <y>0</y>
     <width>598</width>
-    <height>222</height>
+    <height>0</height>
```
Comment (mderasse) [inline src/qt/forms/askpassphrasedialog.ui:10]: ?

File: src/qt/forms/askpassphrasedialog.ui [hunk 5]
```diff
@@ -29,6 +29,31 @@
...
+     <property name="enabled">
+      <bool>false</bool>
+     </property>
+     <property name="text">
+     <string/>
+     </property>
+     <property name="icon">
+      <iconset>
+        <normaloff>:/icons/verium</normaloff>
+        <disabledoff>:/icons/verium</disabledoff>:/icons/verium
```
Comment (mderasse) [inline src/qt/forms/askpassphrasedialog.ui]: Weird syntax no ?

File: src/qt/loginoverlay.cpp [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+    setWindowFlags(Qt::WindowStaysOnTopHint);
+    if (parent) {
+        parent->installEventFilter(this);
+        raise();
+    }
+    setVisible(true);
+    // apply style
+    // XXX: Use local path for development
+    // XXX: To remove before for release
+    QString strPath(QCoreApplication::applicationDirPath() + "/res/style.qss");
```
Comment (mderasse) [inline src/qt/loginoverlay.cpp]: Needed ?

File: .vscode/settings.json (hunk 1 above)
Comment (effectsToCause) [reply .vscode/settings.json]: yes will remove and add to gitignore

File: src/qt/askpassphrasedialog.cpp (hunk 2 above)
Comment (effectsToCause) [reply src/qt/askpassphrasedialog.cpp]: I thought is was needed at the time but maybe not will test this and remove if possible

File: src/qt/askpassphrasedialog.cpp (hunk 3 above)
Comment (effectsToCause) [reply src/qt/askpassphrasedialog.cpp:116]: :)

File: src/qt/forms/askpassphrasedialog.ui (hunk 4 above)
Comment (effectsToCause) [reply src/qt/forms/askpassphrasedialog.ui:10]: Purely for asthetics I wanted to have the askpassphrasedialog only be the height it needs to be so I removed the fixed height min.

File: src/qt/forms/askpassphrasedialog.ui (hunk 5 above)
Comment (effectsToCause) [reply src/qt/forms/askpassphrasedialog.ui]: yes that was how another static icon in modaloverlay was implemented and I thought I we may keep it for some flexibility with usage but we could probably just keep it simple.

File: src/qt/loginoverlay.cpp (hunk 6 above)
Comment (effectsToCause) [reply src/qt/loginoverlay.cpp]: will check

**Comment by effectsToCause** [comment]
will add a commit or two

File: src/qt/askpassphrasedialog.cpp (hunk 2 above)
Comment (mderasse) [reply src/qt/askpassphrasedialog.cpp]: Since Bitcoingui load it, it should be enough :)

File: src/qt/forms/askpassphrasedialog.ui (hunk 4 above)
Comment (mderasse) [reply src/qt/forms/askpassphrasedialog.ui:10]: ok

**Review by mderasse — APPROVED** [review]

**Pull request merged**